cloud/cloudstack/cs_ip_address.py

```diff
@@ -0,0 +1,459 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: cs_ip_address
+short_description: Manages Public/Secondary IP address associations
+description:
+    - Acquires and associates a public IP to an account. Due to API limitations,
+    - this is not an idempotent call, so be sure to only conditionally call this
+    - when C(state=present)
+version_added: '2.0'
+author: "Darren Worrall @dazworrall"
+options:
+  ip_address:
+    description:
+      - Public IP address.
+      - Required if C(state=absent)
+    required: true
+  domain:
+    description:
+      - Domain the IP address is related to.
+    required: false
+    default: null
+  network:
+    description:
+      - Network the IP address is related to.
+    required: false
+    default: null
+  account:
+    description:
+      - Account the IP address is related to.
+    required: false
+    default: null
+  project:
+    description:
+      - Name of the project the IP address is related to.
+    required: false
+    default: null
+  zone:
+    description:
+      - Name of the zone in which the virtual machine is in.
+      - If not set, default zone is used.
+    required: false
+    default: null
+  poll_async:
+    description:
+      - Poll async jobs until job has finished.
+    required: false
+    default: true
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Associate an IP address
+- local_action:
+    module: cs_ip_address
+    account: My Account
+  register: ip_address
+  when: create_instance|changed
+
+# Disassociate an IP address
+- local_action:
+    module: cs_ip_address
+    ip_address: 1.2.3.4
+    state: absent
+'''
+
+RETURN = '''
+---
+ip_address:
+  description: Public IP address.
+  returned: success
+  type: string
+  sample: 1.2.3.4
+zone:
+  description: Name of zone the IP address is related to.
+  returned: success
+  type: string
+  sample: ch-gva-2
+project:
+  description: Name of project the IP address is related to.
+  returned: success
+  type: string
+  sample: Production
+account:
+  description: Account the IP address is related to.
+  returned: success
+  type: string
+  sample: example account
+domain:
+  description: Domain the IP address is related to.
+  returned: success
+  type: string
+  sample: example domain
+'''
+
+
+try:
+    from cs import CloudStack, CloudStackException, read_config
+    has_lib_cs = True
+except ImportError:
+    has_lib_cs = False
+
+# import cloudstack common
+class AnsibleCloudStack:
```

Comment:
This class should be replaced with 
~~~
from ansible.module_utils.cloudstack import *
~~~
in official ansible extras modules. 

I know it is not that easy to test it and I use it this way in inofficial https://github.com/resmo/ansible-cloudstack.